Present AppSSO on the top-most presentedViewController

pascoej opened the pull request:
#### 19987381048be471e9b1bbf2b676a97c95098fbd
<pre>
Present AppSSO on the top-most presentedViewController
<a href="https://bugs.webkit.org/show_bug.cgi?id=274028">https://bugs.webkit.org/show_bug.cgi?id=274028</a>
<a href="https://rdar.apple.com/127761880">rdar://127761880</a>

Reviewed by Aditya Keerthi.

In some situations AppSSO will be called where the result of [m_page-&gt;cocoaView() window].rootViewController
is not applicable to present a view controller on top of because it&apos;s already presenting something. To fix
this we can look at the top-most presentedViewController of [m_page-&gt;cocoaView() window].rootViewController.

* Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
(WebKit::SOAuthorizationSession::presentViewController):

Canonical link: <a href="https://commits.webkit.org/278715@main">https://commits.webkit.org/278715@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/535ccf1dfc50d410e318c13db828d08f95a20c9e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51366 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30672 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3710 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54629 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/2056 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36987 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1735 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/41844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53465 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28316 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44292 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22955 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25645 "Passed tests") | | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1627 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56219 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26483 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1518 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/49241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27724 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44357 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/48420 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11241 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28615 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27459 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
```diff
@@ -399,6 +399,8 @@ static bool isSameOrigin(const WebCore::ResourceRequest& request, const WebCore:
     if (!presentingViewController)
         presentingViewController = [m_page->cocoaView() window].rootViewController;
+    while (presentingViewController && presentingViewController.presentedViewController)
+        presentingViewController = presentingViewController.presentedViewController;
```
Seems like we could use `_wk_viewControllerForFullScreenPresentation` here.

pxlcoder commented on Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
```diff
@@ -398,7 +399,7 @@ static bool isSameOrigin(const WebCore::ResourceRequest& request, const WebCore:
 #if !PLATFORM(VISION)
     if (!presentingViewController)
-        presentingViewController = [m_page->cocoaView() window].rootViewController;
+        presentingViewController = m_page->cocoaView().get()._wk_viewControllerForFullScreenPresentation;
```
Nit:
```suggestion
        presentingViewController = [m_page->cocoaView() _wk_viewControllerForFullScreenPresentation];
```

pascoej committed:
Present AppSSO on the top-most presentedViewController
https://bugs.webkit.org/show_bug.cgi?id=274028
rdar://127761880

Reviewed by Aditya Keerthi.

In some situations AppSSO will be called where the result of [m_page->cocoaView() window].rootViewController
is not applicable to present a view controller on top of because it's already presenting something. To fix
this we can look at the top-most presentedViewController of [m_page->cocoaView() window].rootViewController.

* Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
(WebKit::SOAuthorizationSession::presentViewController):

Canonical link: https://commits.webkit.org/278715@main
